src/main/java/Lab1.java

```diff
@@ -0,0 +1,126 @@
+// Importing required classes
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner; // Changed to Scanner
+
+// Class to manage a to-do list
+public class Lab1 {
+
+    // Main method to run the program
+    public static void main(String[] args) {
+
+        // Initialize variables
+        List<String> tasks = new ArrayList<>();         // List for task descriptions (String type)
+        List<Boolean> completed = new ArrayList<>();    // List for completion status
+        Scanner scanner = new Scanner(System.in);       // Use Scanner for input
+        boolean continueAdding = true;                  // Boolean flag to control loop
+
+        // Print welcome message
+        System.out.println("Welcome to the To-Do List Manager!");
+
+        // Loop to collect tasks
+        while (continueAdding) {
+
+            // Get user input for task description
+            System.out.print("Enter task description (or 'quit' to exit): ");
+            String task = scanner.nextLine();  // Changed from IO.readln()
+
+            // Check if user wants to quit
+            if (task.equalsIgnoreCase("quit")) {
+                continueAdding = false;
+                continue;
+            }
+
+            // Validate task input
+            if (task.trim().isEmpty()) {
+                System.out.println("Error: Task cannot be empty.");
+                continue;
+            }
+
+            // Ask if task is completed
+            System.out.print("Is this task completed? (yes/no): ");
+            String statusInput = scanner.nextLine().toLowerCase();  // Changed from IO.readln()
+
+            boolean isCompleted;
+
+            // Handle potential input errors
+            try {
+
+                // Validate completion status for 'yes'
+                if (statusInput.equals("yes")) {
+                    isCompleted = true;
+
+                    // Validate completion status for 'no'
+                } else if (statusInput.equals("no")) {
+                    isCompleted = false;
+
+                    // Handle invalid completion status
+                } else {
+                    throw new IllegalArgumentException("Please enter 'yes' or 'no'.");
+                }
+
+                // Add task and status to lists
+                tasks.add(task);
+                completed.add(isCompleted);
+                System.out.println("Added task: " + task + " (Completed: " + isCompleted + ")");
+
+                // Catch invalid input errors
+            } catch (IllegalArgumentException e) {
+                System.out.println("Error: " + e.getMessage());
+            }
```

Comment:
FYI, in this scenario there isn't really a need to throw and catch an error.  SImpler to just print the error message directly instead of throwing, and then follow the print with a `continue`